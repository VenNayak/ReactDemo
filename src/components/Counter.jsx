import React from 'react';
import { AppContext } from '../context/AppContext';
class Counter extends React.Component {

    constructor(props) {
        super(props);
        this.inc = this.inc.bind(this);
        //  this.dec = this.dec.bind(this);
        // this.state = {
        //     count : 10
        // }
        this.inputRef = null;

    }


    // inputRef;
    state = {
        count: 10
    }

    change = (evt) => {
        let count = Number(evt.target.value);
        this.setState({ count: count });

    }

    inc() {
        let count = this.state.count + 1;
        this.setState({ count: count });
        console.log("state count :" + this.state.count);
    }

    dec = () => {
        let count = this.state.count - 1;
        this.setState({ count: count });
    }

    update = () => {
        const count = this.inputRef.value ? Number(this.inputRef.value) : this.inputRef.value;
        this.setState({ count: count });
    }

    render() {

        return (
            //return the JSX;
            <div>
                <p>Counter : {this.props.title} -- {this.state.count}</p>
                <p>This is a react class component</p>
                <div>
                    <p>App name : {this.context.appName}</p>
                    <p>User name : {this.context.userName}</p>
                </div>
                <button id="inc" onClick={this.inc}>Increment</button> &nbsp;
        <button onClick={this.dec}> Decrement</button>
                <div>
                    count: <input type="number" onChange={this.change} value={this.state.count} />
                    <div>
                        count: <input type="number" ref={r => this.inputRef = r} /> &nbsp;
          <button onClick={this.update}>Update</button>
                    </div>
                </div>
            </div>

        );
    }

}
Counter.contextType = AppContext;
export default Counter;
