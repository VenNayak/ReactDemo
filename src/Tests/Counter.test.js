import React from 'react';
import Counter from '../components/Counter';

import {mount,configure,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ExpansionPanelActions } from '@material-ui/core';
import {create} from 'react-test-renderer';


configure({adapter:new Adapter()});

describe("Counter Tests", () => {

       it("should be created",() => {
             const container = mount(<Counter/>);
             const instance = container.instance();
             
             expect(instance).toBeTruthy();
            

       });

       it("should have state count as 10",() => {
            const container = mount(<Counter/>);       
            const state = container.state();
            expect(state.count).toBe(10);
           

      });

      it("should increment count by 1",() => {
            const container = mount(<Counter/>);
            const instance = container.instance();
            instance.inc();
            const state = container.state();
            expect(state.count).toBe(11);
           

      });

      it("should increment count by 1 on button click",() => {
           // const container = mount(<Counter/>);
            const shallow = mount(<Counter/>); 
            const button = shallow.find("#inc");
            button.simulate("click");
            button.simulate("click");
            button.simulate("click");
            const state = shallow.state();
            expect(state.count).toBe(13);
           

      });

      it("snapshot",() => {
            const container = create(<Counter title = "Testing Counter"/>);
           expect(container.toJSON()).toMatchSnapshot();
      })

})


