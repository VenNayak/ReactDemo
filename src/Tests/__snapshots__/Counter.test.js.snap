// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Tests snapshot 1`] = `
<div>
  <p>
    Counter : 
    Testing Counter
     -- 
    10
  </p>
  <p>
    This is a react class component
  </p>
  <div>
    <p>
      App name : 
      Training
    </p>
    <p>
      User name : 
      Anil
    </p>
  </div>
  <button
    id="inc"
    onClick={[Function]}
  >
    Increment
  </button>
    
  <button
    onClick={[Function]}
  >
     Decrement
  </button>
  <div>
    count: 
    <input
      onChange={[Function]}
      type="number"
      value={10}
    />
    <div>
      count: 
      <input
        type="number"
      />
        
      <button
        onClick={[Function]}
      >
        Update
      </button>
    </div>
  </div>
</div>
`;
